.github/workflows/build-git-annex-macos.yaml

```diff
@@ -0,0 +1,202 @@
+name: Build git-annex snapshot for macOS
+
+on:
+  # Trigger the workflow on pull request,
+  pull_request:
+    paths:
+      - '.github/workflows/build-git-annex-macos.yaml'
+  schedule:
+    - cron: '30 01 * * *'
+
+env:
+  LANG: C
+
+jobs:
+  build-package:
+    runs-on: macos-latest
+    strategy:
+      # Do not fail fast so we could upload full build log and possible source
+      # pkgs even if binary package building fails
+      fail-fast: false
+    steps:
+    - name: Set up system
+      shell: bash
+      run: |
+        brew install libmagic pkg-config
+
+    - name: Checkout this repository
+      uses: actions/checkout@v1
+
+    - name: Building it all
+      run: |
+        scripts/ci/build_git_annex_macos
+
+    - name: Upload packages
+      uses: actions/upload-artifact@v2-preview
+      with:
+          name: git-annex-macos-dmg
+          path: git-annex/tmp/git-annex.dmg
+
+  test-annex:
+    runs-on: macos-latest
+    needs: build-package
+    strategy:
+      matrix:
+        flavor: [normal, crippled-tmp, crippled-home]
+      fail-fast: false
+
+    steps:
+      - name: Download git-annex package
+        uses: actions/download-artifact@v2-preview
+        with:
+          name: git-annex-macos-dmg
+
+      - name: Install git-annex package
+        run: |
+          hdiutil attach git-annex.dmg
+          rsync -a /Volumes/git-annex/git-annex.app /Applications/
+          hdiutil detach /Volumes/git-annex/
+
+      - name: Add git-annex to PATH
+        run: |
+          echo "::add-path::/Applications/git-annex.app/Contents/MacOS"
+
+      - name: Checkout this repository for scripts/ci
```

Comment:
Please add here a step which would print the version since ATM we follow original build close and `git-annex.dmg` does not include version in its filename (unlike [what we have for .deb packages](https://github.com/datalad/datalad-extensions/runs/1059349934?check_suite_focus=true#step:3:9), so it is possible to discover version really easy).  FWIW -- we could have that step in both linux and osx workflow. `git annex version` output could be generally useful since summarizes build params etc

Reply:
Done.